vllm/model_executor/model_loader/utils.py

```diff
@@ -37,6 +50,32 @@ def get_model_architecture(
             and "MixtralForCausalLM" in architectures):
         architectures = ["QuantMixtralForCausalLM"]
 
+    vllm_supported_archs = ModelRegistry.get_supported_archs()
+    for i, arch in enumerate(architectures):
+        if arch == "TransformersModel":
+            continue
+        custom_module = None
+        if hasattr(model_config.hf_config, "auto_map"):
+            custom_module = transformers.dynamic_module_utils.get_class_from_dynamic_module(
+                model_config.hf_config.auto_map["AutoModel"],
+                model_config.model)
+        if model_config.model_impl == ModelImpl.TRANSFORMERS:
+            if not is_transformers_impl_compatible(arch, custom_module):
+                raise ValueError(
+                    "The Transformers implementation of %s is not compatible "
+                    "with vLLM.", arch)
+            architectures[i] = "TransformersModel"
+        if (model_config.model_impl == ModelImpl.AUTO
+                and arch not in vllm_supported_archs):
+            if not is_transformers_impl_compatible(arch, custom_module):
+                raise ValueError(
+                    "%s has no vLLM implementation and the Transformers "
+                    "implementationis not compatible with vLLM.", arch)
+            logger.info(
+                "%s has no vLLM implementation, falling back to "
+                "Transformers implementation", arch)
```

Comment:
I think this is worth elevating as a warning and including a note that performance+feature support may be degraded

Reply:
Can't mark as resolved myself (not my PR), but this should be done now.